lua/weapons/weapon_flare.lua

```diff
@@ -51,34 +51,36 @@ function SWEP:PrimaryAttack()
     self:EmitSound("Weapon_StunStick.Swing", nil, 60, 0.2)
 
     ply:TimerSimple(0.15, function()
-        local dir = ply:GetAimVector()
-        local org = ply:GetShootPos() + dir * 20 + Vector(0, 0, -10)
-        local effectdata = EffectData()
-        effectdata:SetOrigin(org)
-        effectdata:SetNormal(dir)
-        effectdata:SetRadius(1)
-        effectdata:SetMagnitude(1)
-        effectdata:SetScale(0.1)
-        util.Effect("ElectricSpark", effectdata)
-
-        --[[ local hit = self.Owner:GetEyeTrace()
+        if (IsValid(ply)) then
```

Comment:
Have you seen this error happen? That seems like a bug in the entitytimer system if so

Reply:
there was a report about a null entity in the flare swep on the server but it said it was on line 3. i wasn't sure where else it could have occurred

Reply:
Through serverdebug?

Reply:
yeah

Reply:
I think the problem is actually the `self` becoming invalid, because the ply:TimerSimple won't callback if ply is invalid.